subsys/nrf_security/CMakeLists.txt

```diff
@@ -43,20 +43,19 @@ if(CONFIG_BUILD_WITH_TFM OR CONFIG_PSA_SSF_CRYPTO_CLIENT)
   # We enable either TF-M or the SSF client PSA crypto interface but we are
   # not in the secure image build
 
-  # Add replacement platform.c for NS build
-  list(APPEND src_zephyr
-    ${ARM_MBEDTLS_PATH}/library/platform.c
-  )
-
   # The current version of the mbed TLS deliverables requires mbedcrypto built
   # and linked in the NS image (e.g. for mbedtls and mbedx509 library).
   # If CC3XX_BACKEND is enabled, configurations need to be converted to
   # OBERON_BACKEND (enabled by default).
 
   get_cmake_property(all_vars VARIABLES)
 
-  # 1. Non-secure should not build the PSA core or drivers
-  set(CONFIG_MBEDTLS_PSA_CRYPTO_C               False)
+  # 1. TF-M enabled samples should not enable the PSA core or drivers
+  #    We still need to state CONFIG_MBEDTLS_PSA_CRYPTO_C for 
+  #    nRF54H20 devices which is not the same as an NS build...
+  if(NOT CONFIG_PSA_SSF_CRYPTO_CLIENT)
+    set(CONFIG_MBEDTLS_PSA_CRYPTO_C               False)
```

Comment:
Why don't we have this rather as Kconfig logic?

Reply:
Currently setting `MBEDTLS_PSA_CRYPTO_C` is a signal that you want to build PSA crypto APIs in your system

This configuration is locally disabled both for NS image and for the nRF54H20 device builds (and `MBEDTLS_PSA_CRYPTO_CLIENT` is still kept enabled)

Reply:
This is not introduced in this PR, but IMO this is not a good pattern... Changing the values of Kconfig options in CMake logic smells bad.
If we want to differentiate between NS/S image it feels like that should be done in a clearer way (use a different Kconfig option, maybe add a new one that conveys the meaning and separates the concerns?).